pkgs/top-level/all-packages.nix

```
@@ -11306,6 +11306,8 @@ with pkgs;
 
   dodgy = with python3Packages; toPythonApplication dodgy;
 
+  dovecot_2_3 = callPackage ../by-name/do/dovecot/2_3.nix { };
```

```suggestion
  dovecot_2_3 = callPackage ../by-name/do/dovecot/2_3.nix { };
  dovecot_2_4 = callPackage ../by-name/do/dovecot/2_4.nix { };
```

You'll need to manually import every version with these multi-version packages unfortunately.

---

do we want `legacyPackages.<arch>.dovecot` (without version) to default to `2_4` *and* have a separate 2_4 package?

I was thinking just default to 2_4 and have a separate 2_3 that we'll set as the default for people on stateVersion 25.05 and below (assuming I get this done in time :p )...

---

Yes, pkgs.dovecot as the latest version (at least in unstable) and also have dovecot_2_4, so people can specify 2.4 explicitly.

---

ah yeah makes sense, will do!

---

sorry, completely missed this one earlier

ig having an explicit 2.4 is nice, but only if we're expecting a 2.5 to roll around soon-ish -- which we aren't.

though the fact that I can't think of a usecase doesn't mean that there isn't one, and we're talking about 4 lines in `all-packages.nix`...

if someone wants to jump in and make the call, be my guest -- I'm fine with it either way.